addons/stock/models/stock_move.py

```diff
@@ -181,6 +180,9 @@ def _default_group_id(self):
     next_serial_count = fields.Integer('Number of SN')
     orderpoint_id = fields.Many2one('stock.warehouse.orderpoint', 'Original Reordering Rule', check_company=True)
     json_forecast = fields.Char('JSON data for the forecast widget', compute='_compute_json_forecast')
+    lot_ids = fields.Many2many('stock.production.lot', compute='_lot_ids_compute', inverse='_lot_ids_inverse', string='Serial Numbers', readonly=False)
```

Comment:
We have some convention about variable and method names (you can see all of them here: https://www.odoo.com/documentation/13.0/reference/guidelines.html#symbols-and-conventions).

So i would use _compute_lot_ids() and _set_lot_ids(). 
Also those methods should be insert in a specific location into the file: 

1. Default method and _default_get
2. Field declarations
3. Compute, inverse and search methods in the same order as field declaration
4. Selection method (methods used to return computed values for selection fields)
5. Constrains methods (@api.constrains) and onchange methods (@api.onchange)
6. CRUD methods (ORM overrides)
7. Action methods
8. And finally, other business methods.

Not all the files are perfect but we try to stick to the guidelines for new task

Reply:
For the method order you can put 
the compute and inverse on lot_ids right after _compute_product_qty()
the onchange after onchange_product_uom()